### DISCUSSION TITLE
"The trusted domain rule works with Rethink's DNS only" - Why?

### DISCUSSION THREAD

**Discussion by refactoriel** [celzero/rethink-app#959] [General]
Per-app trusting specific domains works well. But I was wondering why this only works with Rethink's DNS. Isn't the app-specific filtering logic implemented in the rethink-app?

Sorry if this has been answered before - I couldn't find it in the issues or the FAQ.

**Comment by lessload** [comment]
Seem like they plain to make it as paid feature. if it true i will use Adguard instead.
In term of privacy, force user to use there own dns is suck ideas.

**Comment by refactoriel** [comment]
The lack of response from the developers is at least a little suspicious, tbh.

**Comment by hussainmohd-a** [comment]
Hi @DanBenHa, @lessload 
The reason why Trust Domain Rules work properly with Rethink DNS is that we cannot control other upstream DNS servers if they are blocked by the upstream provider. Consequently, users might perceive that Trust Domain Rules are not functioning as intended. However, with Rethink DNS, we can still resolve trusted domains even if they are included in the upstream blocklist. This issue will be addressed once we introduce a new user interface, allowing users to select a specific DNS server solely for resolving trusted domains.

**Reply by refactoriel** [reply]
Thanks for the reply! Could you maybe give a high-level overview of the technical implementation? Do you send the requesting app's name to your DNS server? Is this something that is supported by the DNS protocol or are you using an additional channel to send that information?

**Reply by hussainmohd-a** [reply]
With Rethink Resolver, if a domain is *trusted* by the user but blocked by the Rethink DNS+ (due to certain blocklists), Rethink DNS+ will resolve the domain anyway (as if there were no blocklists present). That is, domains that are blocked by Rethink DNS+ but *trust*ed by the user will still get resolved. That's our specific implementation.

However, this behavior may not occur if the user has set a different upstream DNS server. That's why there is a warning message to let the user know that there can be cases where trusted domains won't behave as intended with other upstream resolvers. No, Rethink does not send any other data regarding the request other than DNS query.

**Reply by refactoriel** [reply]
I see, thanks. So it's only a warning and the logic is implemented in the rethink-app. Just verified it using another DNS server as default.

**Reply by ignoramous** [reply]
Also: https://github.com/celzero/rethink-app/issues/874#issuecomment-1553585805
